src/app/components/team/Statuses/EditStatusDialog.js

```diff
@@ -0,0 +1,187 @@
+    >
+      <DialogTitle>
+        { status ? (
+          <FormattedMessage
+            id="editStatusDialog.titleEdit"
+            defaultMessage="Edit status"
+          />
+        ) : (
+          <FormattedMessage
+            id="editStatusDialog.title"
+            defaultMessage="Add a new status"
+          />
+        )}
+      </DialogTitle>
+      <DialogContent>
+        <TextField
+          id="edit-status-dialog__status-name"
+          inputProps={{
+            maxLength,
+          }}
+          label={(
+            <FormattedMessage
+              id="editStatusDialog.statusTitle"
+              defaultMessage="Status ({maxLength} characters max)"
+              values={{ maxLength }}
+            />
+          )}
+          value={statusName}
+          onChange={e => setStatusName(e.target.value)}
+          variant="outlined"
+          margin="normal"
+          fullWidth
+        />
+        <TextField
+          id="edit-status-dialog__status-description"
+          label={(
+            <FormattedMessage
+              id="editStatusDialog.statusDescription"
+              defaultMessage="Description"
+            />
+          )}
+          value={statusDescription}
+          onChange={e => setStatusDescription(e.target.value)}
+          variant="outlined"
+          margin="normal"
+          fullWidth
+        />
+        <StyledColorPickerContainer>
+          <StyledSelectedColor>
+            <StyledParamName>
+              <FormattedMessage
+                id="editStatusDialog.statusColor"
+                defaultMessage="Status color"
+              />
+            </StyledParamName>
+            <StyledHexValue>{statusColor}</StyledHexValue>
+            <StyledSelectedColorDisplay color={statusColor} />
+          </StyledSelectedColor>
+          <StyledSliderContainer>
+            <SliderPicker
+              color={statusColor}
+              onChangeComplete={color => setStatusColor(color.hex)}
+            />
+          </StyledSliderContainer>
+        </StyledColorPickerContainer>
+      </DialogContent>
+      <DialogActions>
+        <Button className="edit-status-dialog__dismiss" onClick={onDismiss}>
+          <FormattedMessage {...globalStrings.cancel} />
+        </Button>
+        <Button
+          className="edit-status-dialog__submit"
+          onClick={handleSubmit}
+          color="primary"
+          variant="contained"
+        >
+          { status ? (
+            <FormattedMessage {...globalStrings.save} />
+          ) : (
+            <FormattedMessage
+              id="editStatusDialog.addButton"
+              defaultMessage="Add status"
+            />
+          )}
+        </Button>
+      </DialogActions>
+    </Dialog>
+  );
+};
+
+EditStatusDialog.propTypes = {
+  defaultLanguage: PropTypes.string.isRequired,
+  onDismiss: PropTypes.func.isRequired,
+  onSubmit: PropTypes.func.isRequired,
+  open: PropTypes.bool.isRequired,
+  status: PropTypes.object,
```

Comment:
The interplay between this and state is unclear to me. It took me a lot of reading to figure out that we're treating this as an uncontrolled component.

Should we rename `status` to `defaultValue`? That would make it all clear.